File: matlab/autd3/PlaneWave.m
%{
%File: PlaneWave.m
%Project: autd3
%Created Date: 10/06/2022
%Author: Shun Suzuki
%-----
%Last Modified: 10/06/2022
%Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
%-----
%Copyright (c) 2022 Shun Suzuki. All rights reserved.
%
%}

classdef PlaneWave < Gain

    methods

        function obj = PlaneWave(varargin)
            obj = obj@Gain();
            d = varargin{1};

            if nargin < 2
                amp = 1.0;
            else
                amp = varargin{2};
            end

            pp = libpointer('voidPtrPtr', obj.ptr);
            calllib('autd3capi', 'AUTDGainPlaneWave', pp, d(1), d(2), d(3), amp);
        end

    end

end

File: dotnet/cs/example/Samples/PlaneWave.cs
/*
 * File: PlaneWave.cs
 * Project: Samples
 * Created Date: 13/09/2023
 * Author: Shun Suzuki
 * -----
 * Last Modified: 14/11/2023
 * Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
 * -----
 * Copyright (c) 2023 Shun Suzuki. All rights reserved.
 * 
 */

using AUTD3Sharp;
using AUTD3Sharp.Gain;
using AUTD3Sharp.Modulation;
using AUTD3Sharp.Utils;

namespace Samples;

internal static class PlaneWaveTest
{
    public static async Task Test<T>(Controller<T> autd)
    {
        var config = new Silencer();
        await autd.SendAsync(config);

        var m = new Sine(150); // AM sin 150 Hz

        var dir = Vector3d.UnitZ;
        var g = new Plane(dir);

        await autd.SendAsync((m, g));
    }
}
